ion/parse_text_simple.go

```diff
@@ -56,15 +57,43 @@ func (p *parser) parseBinary(annotations []Symbol) Value {
 func (p *parser) parseLongString(annotations []Symbol) String {
 	text := []byte{}
 	for item := p.peekNonComment(); item.Type == lex.IonStringLong; item = p.peekNonComment() {
-		text = append(text, doStringReplacements(p.next().Val)...)
+		text = append(text, p.doStringReplacements(p.next().Val)...)
 	}
 
 	return String{annotations: annotations, text: text}
 }
 
+// decodeHex takes a slice that is a hex-encoded rune (up to 4 bytes large)
+// and decodes it into a slice containing UTF-8 code units.
+func (p *parser) decodeHex(hex []byte) []byte {
+	// must be a power of two and no larger that uint32
+	hexLen := len(hex)
+	if hexLen <= 0 || hexLen > 8 || hexLen%2 != 0 {
+		// calling code must give us a proper slice...
+		p.panicf("Unexpected hex slice handed to decoder: %v", hex)
```

Comment:
Nitpick: This message doesn't give me any hints as to what constraints were violated.  The comment on line 69 seems like a pretty good start =)

Reply:
Done.